docs/sources/getting-started/ReactiveUI-101.md

```diff
@@ -26,90 +29,154 @@ public MainWindow()
+    {
+        get { return _SearchResults.Value; }
+    }
 
-            if (doc.Root == null)
-                return null;
+    // Here, we want to create a property to represent when the application 
+    // is performing a search (i.e. when to show the "spinner" control that 
+    // lets the user know that the app is busy). We also declare this property
+    // to be the result of an Observable (i.e. its value is derived from 
+    // some other property)
 
-            var titles = doc.Root.Descendants("{http://search.yahoo.com/mrss/}title")
-                .Select(x => x.Value);
+    ObservableAsPropertyHelper<Visibility> _SpinnerVisibility;
+    public Visibility SpinnerVisibility
+    {
+        get { return _SpinnerVisibility.Value; }
+    }
 
-            var tagRegex = new Regex("<[^>]+>", RegexOptions.IgnoreCase);
-            var descriptions = doc.Root.Descendants("{http://search.yahoo.com/mrss/}description")
-                .Select(x => tagRegex.Replace(HttpUtility.HtmlDecode(x.Value), ""));
+    public AppViewModel()
+    {
+        ExecuteSearch = ReactiveCommand.CreateAsyncTask(parameter => GetSearchResultsFromFlickr(this.SearchTerm));
 
-            var items = titles.Zip(descriptions,
-                (t, d) => new FlickrPhoto { Title = t, Description = d }).ToArray();
+        /* Creating our UI declaratively
+         * 
+         * The Properties in this ViewModel are related to each other in different 
+         * ways - with other frameworks, it is difficult to describe each relation
+         * succinctly; the code to implement "The UI spinner spins while the search 
+         * is live" usually ends up spread out over several event handlers.
+         *
+         * However, with RxUI, we can describe how properties are related in a very 
+         * organized clear way. Let's describe the workflow of what the user does in
+         * this application, in the order they do it.
+         */
 
-            var urls = doc.Root.Descendants("{http://search.yahoo.com/mrss/}thumbnail")
-                .Select(x => x.Attributes("url").First().Value);
+        // We're going to take a Property and turn it into an Observable here - this
+        // Observable will yield a value every time the Search term changes (which in
+        // the XAML, is connected to the TextBox). 
+        //
+        // We're going to use the Throttle operator to ignore changes that 
+        // happen too quickly, since we don't want to issue a search for each 
+        // key pressed! We then pull the Value of the change, then filter 
+        // out changes that are identical, as well as strings that are empty.
+        //
+        // Finally, we use RxUI's InvokeCommand operator, which takes the String 
+        // and calls the Execute method on the ExecuteSearch Command, after 
+        // making sure the Command can be executed via calling CanExecute.
+        this.ObservableForProperty(x => x.SearchTerm)
```

Comment:
I think we should :fire: any documented usages of `ObservableForProperty` in favour of `WhenAny`/ `WhenAnyValue`. Thoughts?